src/databricks/labs/ucx/hive_metastore/locations.py

```diff
@@ -248,3 +250,107 @@ def _try_fetch(self) -> Iterable[Mount]:
+        """
+        return self._snapshot(partial(self._try_load), partial(self._crawl))
+
+    def _try_load(self) -> Iterable[Table]:
+        """Tries to load table information from the database or throws TABLE_OR_VIEW_NOT_FOUND error"""
+        for row in self._fetch(f"SELECT * FROM {escape_sql_identifier(self.full_name)}"):
+            yield Table(*row)
+
+    def _crawl(self):
+        all_mounts = self._mc.snapshot()
+        all_tables = []
+        for mount in all_mounts:
+            if self._include_mounts and mount.name not in self._include_mounts:
+                continue
+            table_paths = self._find_delta_log_folders(mount.name)
+            for path, entry in table_paths.items():
+                table = Table(
+                    catalog="hive_metastore",
+                    database="tables_in_mounts",
+                    name=self._get_table_name(path),
+                    object_type="EXTERNAL",
+                    table_format="DELTA" if entry.is_delta else "PARQUET",
+                    location=path,
+                )
+                all_tables.append(table)
+        return all_tables
+
+    def _find_delta_log_folders(self, root_dir, delta_log_folders=None) -> dict:
+        if delta_log_folders is None:
+            delta_log_folders = {}
+
+        entries = self._dbutils.fs.ls(root_dir)
+        for entry in entries:
+            if self._is_irrelevant(entry.name):
+                continue
+
+            path = self._get_absolute_directory(entry.path)
+
+            if self._is_parquet(entry.path):
+                if self._path_is_delta(delta_log_folders, path):
+                    continue
+                delta_log_folders[path] = TableInMount(is_delta=False, is_parquet=True, is_partitioned=False)
+            elif entry.name == "_delta_log":
+                logger.debug(f"Found delta table {path}")
+                delta_log_folders[path] = TableInMount(is_delta=True, is_parquet=False, is_partitioned=False)
+            else:
+                self._find_delta_log_folders(entry.path, delta_log_folders)
+
+        return delta_log_folders
+
+    def _get_absolute_directory(self, path):
+        """
+        Returns the absolute directory of a given path.
+        Example: _get_absolute_directory('/mnt/test/123/dxs/aaa.parquet') -> /mnt/test/123/dxs/
+        """
+        return re.search(r'^(.*/)[^/]+/?', path).group(1)
+
+    def _get_table_name(self, path):
+        """
+        Returns the parent directory of a given path.
+        Example: _get_table_name('/mnt/test/123/dxs/aaa.parquet') -> dxs
+        """
+        return re.search("([^/]+)/?$", path, re.IGNORECASE).group(1)
+
+    def _path_is_delta(self, delta_log_folders, path):
+        return delta_log_folders.get(path, {}) and delta_log_folders.get(path, {}).is_delta
+
+    def _is_parquet(self, entry: str) -> bool:
+        parquet_patterns = {'.parquet'}
```

Comment:
```suggestion
        parquet_patterns = {'.parquet', '='}
```
this way you'll detect partitions OR non-partitioned tables.

Reply:
Partition support should also work for delta tables, not parquet files only. And I would like to avoid scanning inside partitions to reduce the listing calls